mmdet3d/models/dense_heads/groupfree3d_head.py

```diff
@@ -0,0 +1,1000 @@
+import copy
+import numpy as np
+import torch
+from mmcv import ConfigDict
+from mmcv.cnn.bricks.transformer import build_transformer_layer
+from mmcv.runner import force_fp32
+from torch import nn as nn
+from torch.nn import functional as F
+
+from mmdet3d.core.post_processing import aligned_3d_nms
+from mmdet3d.models.builder import build_loss
+from mmdet3d.ops import Points_Sampler, gather_points
+from mmdet.core import build_bbox_coder, multi_apply
+from mmdet.models import HEADS
+from .base_conv_bbox_head import BaseConvBboxHead
+
+
+class PointsObjClsModule(nn.Module):
+    """object candidate point prediction from seed point features.
+
+    Args:
+        seed_feature_dim (int): number of channels of seed point features
+    """
+
+    def __init__(self, seed_feature_dim):
+        super().__init__()
+        self.in_dim = seed_feature_dim
+        self.conv1 = torch.nn.Conv1d(self.in_dim, self.in_dim, 1)
+        self.bn1 = torch.nn.BatchNorm1d(self.in_dim)
+        self.conv2 = torch.nn.Conv1d(self.in_dim, self.in_dim, 1)
+        self.bn2 = torch.nn.BatchNorm1d(self.in_dim)
+        self.conv3 = torch.nn.Conv1d(self.in_dim, 1, 1)
+
+    def forward(self, seed_features):
+        """Forward pass.
+
+        Args:
+            seed_features (torch.Tensor): seed features, dims:
+                (batch_size, feature_dim, num_seed)
+
+        Returns:
+            torch.Tensor: objectness logits, dim:
+                (batch_size, 1, num_seed)
+        """
+        net = F.relu(self.bn1(self.conv1(seed_features)))
```

Comment:
We should also use the relu Module instead of the function, so that it can be tracked in the model.